src/Serializer/AbstractSerializer.php

```diff
@@ -191,7 +191,7 @@ protected function serializeObject($object, int $_depth = 0, array $hashes = [])
         $hashes[] = spl_object_hash($object);
         $serializedObject = [];
 
-        foreach ($object as $key => &$value) {
+        foreach (get_object_vars($object) as $key => &$value) {
```

Comment:
Isn't this a BC break?

Reply:
I do not know, unit tests do not break...

Reply:
Iteration over an object belongs to the Bad Parts of PHP.

Reply:
>Iteration over an object belongs to the Bad Parts of PHP.

Indeed, but as I mentioned [it's a BC](https://3v4l.org/NEgUp) so we cannot change it ✋

Reply:
I've tried with `(array)`. It made tests fail :(